cabal.project

```diff
@@ -65,7 +65,10 @@ constraints:
   -- https://github.com/ndmitchell/hlint/issues/1376
   hlint +ghc-lib,
   ghc-lib-parser-ex -auto,
-  stylish-haskell +ghc-lib
+  stylish-haskell +ghc-lib,
+  -- implicit-hie 0.1.3.0 introduced an unexpected behavioral change.
+  -- https://github.com/Avi-D-coder/implicit-hie/issues/50
+  implicit-hie < 0.1.3
```

Comment:
I think this needs to go into `ghcide.cabal`, otherwise the Hackage release will still permit this version

Reply:
Unfortunately, we cannot specify it there. The `ghcide` library depends on `implicit-hie-cradle` only, not `implicit-hie` directly. Though we have the target `ghcide:ghcide-tests` that depends on `implicit-hie:gen-hie`, putting a constraint there does not forbid installing `implicit-hie 0.1.3.0` in my experiment, so I couldn't find a better way than this yet.

Reply:
doesn't adding the depedency `implicit-hie` to ghcide.cabal work, even if the dependency is unused?

Reply:
It probably does. But also we need to try and resolve this upstream. @Avi-D-coder do you know what's up? Should we be deprecating the broken versions so it doesn't get picked by cabal ever?

Reply:
ah. yeah. I refrain myself from adding an unused dep. saw the comment in #3404 now. will wait for merging that fix onto master first as discussed.

Reply:
I have just set `[0.1.3.0](https://hackage.haskell.org/package/implicit-hie-0.1.3.0)` as deprecated.
I have not had the time to finish implementing https://github.com/Avi-D-coder/implicit-hie/issues/50#issuecomment-1316100308.